### PR TITLE
feat: implement MUL opcode

### LINKED ISSUE #5: Opcode: MUL

### PR DESCRIPTION
**Pull request by Danikim01**
Closes #5 
This PR implements the code generation for the MUL opcode

### CODE REVIEW

File: tests/operations.rs [hunk 1]
```diff
@@ -79,3 +79,20 @@ fn push_push_add() {
 fn add_with_stack_underflow() {
     run_program_assert_revert(vec![Operation::Add]);
 }
+
+#[test]
+fn push_push_mul() {
+    let (a, b) = (2, 2);
```
Comment (MegaRedHand) [inline tests/operations.rs:85]: Please make it two different numbers, so we can catch more errors.

**Review by MegaRedHand — COMMENTED** [review]
Good work!! Please merge with main and address comments.

File: tests/operations.rs [hunk 2]
```diff
@@ -79,3 +79,20 @@ fn push_push_add() {
...
+
+    let program = vec![
+        Operation::Push32(new_32_byte_immediate(a)),
+        Operation::Push32(new_32_byte_immediate(b)),
+        Operation::Mul,
+    ];
```
Comment (MegaRedHand) [inline tests/operations.rs:91]: Please add a test for numeric overflow. When both numbers are big enough, the result should wrap around.

**Review by MegaRedHand — CHANGES_REQUESTED** [review]
I forgot about this edge-case.

File: tests/operations.rs (hunk 2 above)
Comment (Danikim01) [reply tests/operations.rs:91]: what do you mean by wrap around?

File: tests/operations.rs [hunk 3]
```diff
@@ -92,6 +92,18 @@ fn push_push_mul() {
     run_program_assert_result(program, a * b);
 }
 
+#[test]
+fn mul_wraps_result() {
+    let a = 128;
+    let b = 2;
+    let program = vec![
```
Comment (MegaRedHand) [inline tests/operations.rs:99]: This example doesn't check u256 overflow, just u8 overflow. For that, we should have a number close to 2^256 - 1, and multiply it by something that overflows it. Note that you'll need to manually represent the number with an array in the operation. Here's an example: https://github.com/lambdaclass/evm_mlir/blob/3ab4e499cca565c5b10f16efc7709e0ee4b98a9f/tests/operations.rs#L106-L119
